themes/bootstrap3/templates/content/markdown.phtml

```diff
@@ -1,7 +1,18 @@
 <?php
-$found = preg_match('/#+(.*)/', $this->data, $matches);
-if ($found) {
-    $this->headTitle($matches[1]);
+$content = $this->markdown($this->data);
+$frontMatter = [];
+$title = '';
+if ($content instanceof League\CommonMark\Extension\FrontMatter\Output\RenderedContentWithFrontMatter) {
+  $frontMatter = $content->getFrontMatter();
+  $title = $frontMatter['title'] ?? '';
+} else {
```

Comment:
What situation causes this else clause to trigger? Will it impact our ability to call `$content->getContent()` below?

Reply:
The else branch is used when FrontMatter extension in disabled. There is no impact on `getContent`. `RenderedContentWithFrontMatter` is subclass of `RenderedContent` which only adds  `getFrontMatter` method. But thanks for pointing on it. In case there is FrontMatter enabled and no title in frontmatter of the page. Then title remains empty. I've changed it to use the first heading as title, just as before this patchset.

Reply:
Great, thanks!